### PR TITLE
Destination snowflake: test tweaks

### PR DESCRIPTION
**Pull request by edgao**
move a test with a hardcoded config out of the base class and into the subclass. It only needs to run once rather than once per subclass.

add some readme notes + disable tests that look annoying to fix.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`cf90d467e9`](https://github.com/airbytehq/airbyte/pull/29402/commits/cf90d467e91f773ecd1625557bfb136e78832222)) - ❌

⏲️  Total pipeline duration: 46mn16s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_snowflake_tests/1692027981/cf90d467e91f773ecd1625557bfb136e78832222/destination-snowflake/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cf90d467e91f773ecd1625557bfb136e78832222)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`5ae1bd57b9`](https://github.com/airbytehq/airbyte/pull/29402/commits/5ae1bd57b9dfd0111232d07272981a1ef895211e)) - ❌

⏲️  Total pipeline duration: 45mn53s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_snowflake_tests/1692033303/5ae1bd57b9dfd0111232d07272981a1ef895211e/destination-snowflake/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5ae1bd57b9dfd0111232d07272981a1ef895211e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`eafe65b2d0`](https://github.com/airbytehq/airbyte/pull/29402/commits/eafe65b2d027a354f8cbb5fa9fa3cb9ac1db2f88)) - ✅

⏲️  Total pipeline duration: 44mn37s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_snowflake_tests/1692036701/eafe65b2d027a354f8cbb5fa9fa3cb9ac1db2f88/destination-snowflake/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:eafe65b2d027a354f8cbb5fa9fa3cb9ac1db2f88)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

File: airbyte-integrations/connectors/destination-snowflake/README.md [hunk 1]
```diff
@@ -93,3 +93,14 @@ DROP USER IF EXISTS INTEGRATION_TEST_USER_DESTINATION;
...
+### Setup for various error-case users:
+Log in as the `INTEGRATION_TEST_USER_DESTINATION` user, and run this:
+```sql
+drop schema if exists INTEGRATION_TEST_DESTINATION.TEXT_SCHEMA;
+create schema INTEGRATION_TEST_DESTINATION.TEXT_SCHEMA;
+grant ownership on schema INTEGRATION_TEST_DESTINATION.TEXT_SCHEMA to role INTEGRATION_TESTER_DESTINATION revoke current grants;
+grant all privileges on schema INTEGRATION_TEST_DESTINATION.TEXT_SCHEMA to role NO_ACTIVE_WAREHOUSE_ROLE;
+```
+
+These tests are currently disabled (`testCheckWithNoProperStagingPermissionConnection`, `testCheckWithNoActiveWarehouseConnection`). Their test users keep breaking (i.e. becoming the schema owner) because our tests are tearing down `TEXT_SCHEMA` after every test.
```
Comment (cynthiaxyin) [inline airbyte-integrations/connectors/destination-snowflake/README.md:106]: TIL this doc exists.

**Review by cynthiaxyin — APPROVED** [review]

**Pull request merged**